apps/settings/js/panels/root/stk_item.js

```diff
@@ -0,0 +1,161 @@
+          callback();
+        });
+    }
+
+    function executeIccCmd(iccMessage) {
+      if (!iccMessage) {
+        return;
+      }
+
+      // Clear cache
+      var reqIccData = window.navigator.mozSettings.createLock().set({
+        'icc.data': null
+      });
+      reqIccData.onsuccess = function icc_getIccData() {
+        window.DUMP('ICC Cache cleared');
+      };
+
+      // Open ICC section
+      window.DUMP('ICC message to execute: ', iccMessage);
+      loadIccPage(function() {
+        var event = new CustomEvent('stkasynccommand', {
+          detail: { 'message': iccMessage }
+        });
+        window.dispatchEvent(event);
+      });
+    }
+
+    setTimeout(function updateStkMenu() {
+      window.DUMP('Showing STK main menu');
+      // XXX https://bugzilla.mozilla.org/show_bug.cgi?id=844727
+      // We should use Settings.settingsCache first
+      var settings = Settings.mozSettings;
+      var lock = settings.createLock();
+
+      function showStkEntries(menu) {
+        window.DUMP('STK cached menu: ', menu);
+        if (!menu || typeof menu !== 'object' ||
+          Object.keys(menu).length === 0) {
+            window.DUMP('No STK available - exit');
+            iccMainHeader.hidden = true;
+            iccEntries.hidden = true;
+            return;
+        }
+
+        // Clean current entries
+        iccEntries.innerHTML = '';
+        iccMainHeader.hidden = true;
+        iccEntries.hidden = true;
+
+        // update and show the entry in settings
+        Object.keys(menu).forEach(function(SIMNumber) {
+          window.DUMP('STK Menu for SIM ' + SIMNumber +
+            ' (' + menu[SIMNumber].iccId + ') - ', menu[SIMNumber].entries);
+
+          var li = document.createElement('li');
+          var a = document.createElement('a');
+          var menuItem = menu[SIMNumber].entries;
+          var icon = STKHelper.getFirstIconRawData(menuItem);
+
+          a.id = 'menuItem-icc-' + menu[SIMNumber].iccId;
+          a.className = 'menu-item menuItem-icc';
+          a.href = '#icc';
+          if (icon) {
+            var iconContainer = document.createElement('span');
+            iconContainer.appendChild(STKHelper.getIconCanvas(icon));
+            li.appendChild(iconContainer);
+            iccEntries.dataset.customIcon = true;
+          } else {
+            a.dataset.icon = 'sim-toolkit';
+          }
+          a.onclick = function menu_icc_onclick() {
+            window.DUMP('Touched ' + menu[SIMNumber].iccId);
+            loadIccPage(function() {
+              var event = new CustomEvent('stkmenuselection', {
+                detail: { 'menu': menu[SIMNumber] }
+              });
+              window.dispatchEvent(event);
+            });
+          };
+
+          var span = document.createElement('span');
+          span.textContent = menu[SIMNumber].entries.title;
+          a.appendChild(span);
+
+          // XXX this line requires a better l10n support, see bug 968853
```

Comment:
we can help to remove this line and use `small.setAttribute('data-l10n-id', 'sim' + SIMNumber)` here